### PR TITLE
Run pytest tests

### PR DESCRIPTION
**Pull request by baszalmstra**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR adds running of the pytest to the recipe test section. This is useful so that we can run downstream tests from dependencies to guard against accidental breakage. 

I also added `pydantic >=2,<3` to the run requirements. This is also the case in the [upstream `environment.yaml`.](https://github.com/conda-forge/conda-smithy/blob/7b0b83038ae6c012c72bc13cca6641fb81dc7e08/environment.yml#L35C5-L35C20)

### CODE REVIEW

**Comment by baszalmstra** [comment]
Ouch that is a lot of errors..

**Comment by baszalmstra** [comment]
I only included specific tests for now. I am unsure why a lot of tests fail but I assume its because the tests do not expect to be run from another repository?

Anyway, I think its still important to be able to run the majority of the tests for downstream tests.

**Comment by baszalmstra** [comment]
@conda-forge-admin, please rerender

**Review by beckermr — CHANGES_REQUESTED** [review]
There is a specific pytest invocation to use for installed packages. We should try that here.

**Comment by baszalmstra** [comment]
> There is a specific pytest invocation to use for installed packages. We should try that here.

Would be happy to! Do you know what the invocation is, a quick google didnt tell me much?

**Comment by baszalmstra** [comment]
> Pydantic is not a runtime requirement afaik. Cc @isuruf who knows this better.

Its used here:

https://github.com/conda-forge/conda-smithy/blob/7b0b83038ae6c012c72bc13cca6641fb81dc7e08/conda_smithy/schema.py#L10

I think thats part of the package.

**Comment by beckermr** [comment]
We currently serialize the schema as json and don't use pydantic at runtime IIRC.

**Comment by baszalmstra** [comment]
> We currently serialize the schema as json and don't use pydantic at runtime IIRC.

But then `import conda_smithy.schema` would just fail right?

**Comment by beckermr** [comment]
True but I don't think that module is ever impoted at runtime when using smithy.

**Comment by beckermr** [comment]
See here:

https://github.com/search?q=repo%3Aconda-forge%2Fconda-smithy%20conda_smithy.schema&type=code

this is by design AFAIK. Other members of @conda-forge/core should confirm or deny.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -69,8 +70,16 @@ test:
...
+  requires:
+    - mock
+    - pytest
+    - pytest-cov
+  source_files:
+    - tests/
   commands:
     - conda smithy --help
+    - pytest tests/test_condaforge_config_schema.py tests/test_ci_skeleton.py tests/test_feedstock_io.py tests/test_feedstock_tokens.py tests/test_lint_recipe.py tests/test_variant_algebra.py
+
```
Comment (beckermr) [inline recipe/meta.yaml:82]: Let's use `--ignore` here so that we don't have to update this list.

**Review by beckermr — CHANGES_REQUESTED** [review]
See the comment above.

**Comment by baszalmstra** [comment]
> See here:
> 
> [github.com/search?q=repo:conda-forge/conda-smithy%20conda_smithy.schema&type=code](https://github.com/search?q=repo%3Aconda-forge%2Fconda-smithy%20conda_smithy.schema&type=code)
> 
> this is by design AFAIK. Other members of @conda-forge/core should confirm or deny.

I wonder why that would be by design because that seems like bad practice. I can imagine it is used just to generate the JSON schema but I dont think it belongs inside the package in that case.

But I would happy to remove it, Ill wait for someone from core to reply.

**Comment by beckermr** [comment]
I do not want to rehash the pydantic issue since it has been an ongoing thing. I doubt this PR will get merged until it is removed.

**Comment by baszalmstra** [comment]
> I do not want to rehash the pydantic issue since it has been an ongoing thing. I doubt this PR will get merged until it is removed.

Ok! I moved it to the test section instead.

**Review by beckermr — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,7 +9,7 @@ source:
   sha256: a4524476bdc93d573f76d7b71ede235a77f61a905ddd820cb1b1124b9276d879
 
 build:
-  number: 0
+  number: 1
```
Comment (beckermr) [inline recipe/meta.yaml:12]:
```suggestion
  number: 0
```

**Review by beckermr — COMMENTED** [review]
We don't need a new build.

**Pull request merged**